dask/optimization.py

```diff
@@ -808,3 +808,155 @@ def key_split(s):
             return result
     except Exception:
         return 'Other'
+
+
+class SubgraphCallable(object):
+    """Create a callable object from a dask graph.
+
+    Parameters
+    ----------
+    dsk : dict
+        A dask graph
+    outkey : hashable
+        The output key from the graph
+    inkeys : list
+        A list of keys to be used as arguments to the callable.
+    name : str, optional
+        The name to use for the function.
+    """
+    __slots__ = ('dsk', 'outkey', 'inkeys', 'name')
+
+    def __init__(self, dsk, outkey, inkeys, name='subgraph_callable'):
+        self.dsk = dsk
+        self.outkey = outkey
+        self.inkeys = inkeys
+        self.name = name
```

Comment:
We don't seem to use the name attribute currently (though I can imagine it being useful)

Reply:
Any response to this?  Should we keep this parameter?

Reply:
I don't see the harm in keeping it. I wanted it for the atop fusion work (so the fused task functions would have nice names).